arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java

```diff
@@ -30,11 +40,80 @@
 @Slf4j
 @RequiredArgsConstructor
 public class ProfitableTransactionSelector implements PluginTransactionSelector {
+  @VisibleForTesting protected static Set<Hash> unprofitableCache = new LinkedHashSet<>();
+  @VisibleForTesting protected static Wei prevMinGasPrice = Wei.MAX_WEI;
+
   private final int verificationGasCost;
   private final int verificationCapacity;
   private final int gasPriceRatio;
   private final double minMargin;
   private final int adjustTxSize;
+  private final int unprofitableCacheSize;
+  private final int unprofitableRetryLimit;
+  private int unprofitableRetries;
```

Comment:
If we maintain a heap sorted by profitability, we might not need this. Because:
* If we're still selecting profitable transactions from this heap, we're doing useful job and should continue
* If we got an unprofitable transaction from the heap, we should just stop, without any retry limits because following transactions will be less profitable

Reply:
And one caveat with profitability heap cache. When gas price decreses, and we're selecting any profitable transaction from profitability cache, we must either select or re-evaluate all of the newly profitable transactions. Otherwise some transactions in this cache that weren't re-evaluated might get stuck with last known bad profitability

Reply:
Another caveat that state in theory may change transaction's profitability (gas consumed)

Reply:
Remember that here we are not in the txpool, and we cannot alter the order in which txs are evaluated, this PR is a quick way to mitigate the impact of unprofitable txs, until we have better solution with txpool that takes in consideration the profitability of a tx, and process txs in that order.
So here what we can do is try to understand if it is worth to retry the execution of a previously unprofitable tx, in case it is again a candidate for inclusion. 
Anyway your suggestion is good to implement a better filter, remembering the _unprofitable_ price for the tx, and re-processing it only if the current min price is lower, we should be able to do better guesses.
If you agree we can deploy this simple mitigation as is, and try to build a better filter in a next PR.

Reply:
Makes sense to me. I think this PR adds some value, so we can deploy it